drivers/cc110x/cc110x_rx_tx.c

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (C) 2018 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup   drivers_cc110x
+ * @{
+ *
+ * @file
+ * @brief     Functions to manage sending/receiving frames with the CC110x
+ *
+ * @author    Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ * @}
+ */
+
+#include "xtimer.h"
+#include "cc110x.h"
+#include "cc110x_internal.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* Use NETDEV_EVENT_ISR to indicate that no event needs to be passed to upper
+ * layer at end of ISR, as ISR will never need this event
+ */
+#define NETDEV_NO_EVENT NETDEV_EVENT_ISR
+
+void cc110x_rx(cc110x_t *dev)
+{
+    //DEBUG("[cc110x] Going to RX\n");
+    /* bring device to IDLE state and flush FIFOs (just in case) */
+    gpio_irq_disable(dev->params.gdo0);
+    gpio_irq_disable(dev->params.gdo2);
+    cc110x_cmd(dev, CC110X_STROBE_IDLE);
+    cc110x_cmd(dev, CC110X_STROBE_FLUSH_RX);
+    cc110x_cmd(dev, CC110X_STROBE_FLUSH_TX);
+    dev->buf.pos = dev->buf.len = 0;
+    /* Apply GDO2 config and go to RX */
+    cc110x_write(dev, CC110X_REG_IOCFG2, CC110X_GDO_ON_RX_DATA);
+    cc110x_write(dev, CC110X_REG_IOCFG0, CC110X_GDO_ON_TRANSMISSION);
+    cc110x_cmd(dev, CC110X_STROBE_RX);
+    dev->state = CC110X_STATE_RX;
+    gpio_irq_enable(dev->params.gdo2);
+    gpio_irq_enable(dev->params.gdo0);
+}
+
+/**
+ * @brief Function to run when frame is fully received
+ *
+ * @param dev     Device descriptor of the transceiver
+ *
+ * Intended to be called from @ref cc110x_rx_continue
+ */
+static netdev_event_t cc110x_rx_done(cc110x_t *dev)
+{
+    uint8_t lqi_crc;
+    int8_t rssi;
+
+    cc110x_read(dev, CC110X_REG_LQI, &lqi_crc);
+    cc110x_read(dev, CC110X_REG_RSSI, (uint8_t *)&rssi);
+
+    /* CRC_OK bit is most significant bit, see page 92 in the data sheet */
+    if (!(lqi_crc & 0x80)) {
+        DEBUG("[cc110x] ISR: CRC error, dropping frame\n");
+        /* Drop frame and go back to RX */
+        cc110x_rx(dev);
+        return NETDEV_EVENT_CRC_ERROR;
+    }
```

Comment:
Should you first check if CRC check is enabled? In some cases the embedded CRC function is not needed and thus disabled. This check would drop the frame even if a valid (different) CRC can be calculated afterwards.

Reply:
This driver does not allow to disable hardware CRC calculation/validation at the moment. If that feature would be added, this would need to be updated as you implied.